API/Async.pm

```diff
@@ -311,20 +322,38 @@ sub playlist {
 # lookup to get the latest timestamp first, then return from cache directly
 # if the list hasn't changed, or look up afresh if needed.
 sub getFavorites {
-	my ($self, $cb, $type, $drill) = @_;
+	my ($self, $cb, $type, $refresh) = @_;
 
 	return $cb->() unless $type;
 
 	my $userId = $self->userId || return $cb->();
 	my $cacheKey = "tidal_favs_$type:$userId";
 
+	# verify if that type has been updated and force refresh (don't confuse adding
+	# a playlist to favorites with changing the *content* of a playlist)
+	if ((my $updated = $self->updated) =~ /$type:/) {
+		$self->updated($updated =~ s/$type://r);
+		$refresh = 1;
+	}
+
 	my $lookupSub = sub {
+		my $timestamp = shift;
+
 		$self->_get("/users/$userId/favorites/$type", sub {
 			my $result = shift;
 
 			my $items = [ map { $_->{item} } @{$result->{items} || []} ] if $result;
 			$items = Plugins::TIDAL::API->cacheTrackMetadata($items) if $items && $type eq 'tracks';
 
+			# verify if home-made playlists need to be invalidated
+			if (defined $timestamp && $type =~ /playlist/) {
+				foreach my $playlist (@$items) {
+					next unless $playlist->{type} =~ /USER/ && str2time($playlist->{lastUpdated}) > $timestamp;
```

Comment:
Can we use exact matches here, instead of the regexes? `type` should always be `playlist` exactly, isn't something like `userplaylist` or similar?

Reply:
it's exact, I used more regex instead of eq recently, not sure why ... lazyness I guess

Reply:
fixed